keepassxc-browser/pwgen.js

```diff
@@ -0,0 +1,383 @@
+'use strict';
+
+var kpxcPassword = {};
+kpxcPassword.observedIcons = [];
+kpxcPassword.observingLock = false;
+kpxcPassword.created = false;
+kpxcPassword.selected = null;
+kpxcPassword.startPosX = 0;
+kpxcPassword.startPosY = 0;
+kpxcPassword.diffX = 0;
+kpxcPassword.diffY = 0;
+kpxcPassword.dialog = null;
+kpxcPassword.titleBar = null;
+
+kpxcPassword.init = function() {
+    if ('initPasswordGenerator' in _called) {
+        return;
+    }
+
+    _called.initPasswordGenerator = true;
+
+    window.setInterval(function() {
+        kpxcPassword.checkObservedElements();
+    }, 400);
+};
+
+kpxcPassword.initField = function(field, inputs, pos) {
+    if (!field) {
+        return;
+    }
+    if (field.getAttribute('kpxc-password-generator')) {
+        return;
+    }
+
+    field.setAttribute('kpxc-password-generator', true);
+
+    kpxcPassword.createIcon(field);
+
+    let found = false;
+    if (inputs) {
+        for (let i = pos + 1; i < inputs.length; i++) {
+            if (inputs[i] && inputs[i].getAttribute('type') && inputs[i].getAttribute('type').toLowerCase() === 'password') {
+                field.setAttribute('kpxc-genpw-next-field-id', inputs[i].getAttribute('kpxc-id'));
+                field.setAttribute('kpxc-genpw-next-is-password-field', (i === 0));
+                found = true;
+                break;
+            }
+        }
+    }
+
+    field.setAttribute('kpxc-genpw-next-field-exists', found);
+};
+
+kpxcPassword.createIcon = function(field) {
+    const className = (isFirefox() ? 'key-moz' : 'key');
+    const size = (field.offsetHeight > 28) ? 24 : 16;
+    let offset = Math.floor((field.offsetHeight - size) / 3);
+    offset = (offset < 0) ? 0 : offset;
+
+    const icon = kpxcUI.createElement('div', 'kpxc kpxc-genpw-icon ' + className,
+        {'title': 'Generate password', 'size': size, 'offset': offset, 'kpxc-genpw-field-id': field.getAttribute('kpxc-id')});
+    icon.style.zIndex = '9999';
+    icon.style.width = String(size) + 'px';
+    icon.style.height = String(size) + 'px';
```

Comment:
a simple string concatenation is totally safe

```js
icon.style.width = size + "px";
```

Reply:
Just wanted to make sure it's clear what's happening here. Also, see [this](https://github.com/airbnb/javascript#type-casting--coercion).